### DISCUSSION TITLE
[Solved] Sourcing an authentic mega2560 pro board.

### DISCUSSION THREAD

**Discussion by VincentEmond** [sanni/cartreader#421] [Need help]
**Solved**

**TLDR;**

Be careful what brand of Mega 2650 pro you buy. Some may come with a couterfeit ATMEL chip that doesn't work according to specs. I can confirm the `ASHATA` brand I got off amazon came with a fake chip. I can also confirm the `Keyestudios` brand that came from Aliexpress came with an authentic chip. If the board is really cheap it's probably a fake. The price for a single Atmega2560 chip is around 20$ USD, sure it might come at less than that if you buy in bulk but, I would go for a board that is maybe 30$ or more, cheaper than that I believe there's a strong chance you get a fake chip.

Hi, 

first and foremost thanks for this awesome project I love open source and I've had a blast building these devices. Currently running a HW3 to preserve my collection and I am loving it!

I figured it was the time to upgrade so I started to build the HW5 version. The problem I have is when I power it on it doesn't work at all, LCD stays dark and nothing happens. I have built it according to the wiki.

First here are some pictures for reference:

![20220721_010854](https://user-images.githubusercontent.com/9730015/180134476-b61ab404-bf29-4ba2-9b50-c4323912d762.jpg)
![20220721_010702](https://user-images.githubusercontent.com/9730015/180134482-1e36c533-9a41-4980-b617-a9d29f74af8b.jpg)
![20220721_010526](https://user-images.githubusercontent.com/9730015/180134489-4eda41d9-1382-4baa-a6e0-026d663e8813.jpg)

**Things to note:** 

- I don't have the Makerbase brand LCD I am using the Mini 12864 from FYSETC [https://wiki.fysetc.com/Mini12864_Panel/](https://wiki.fysetc.com/Mini12864_Panel/)
- It seems like the pinout of EXP1 and EXP2 are the same so it shouldn't make a big difference.
- I modified the LCD to get the 3.3V on the right pin on EXP2, this one doesn't have a jumper unfortunately. My arduino does turn on with 3.3V so this seems like its working. (I also closed the TP3-4 jumper)

![mod](https://user-images.githubusercontent.com/9730015/180133529-dc788b16-caae-459d-8bb9-fbab85f50a0e.jpg)

- I didn't use a dupont wire for the USB 5V connection. I soldered some 24AWG wire directly between the board and the Arduino.
- Here are my defines
```cpp 
//******************************************
// !!! CHOOSE HARDWARE VERSION !!!
//******************************************
// Remove // in front of the line with your hardware version
 #define HW5
// #define HW4
// #define HW3
// #define HW2
// #define HW1 
#define SERIAL_MONITOR

#if !(defined(HW1) || defined(HW2) || defined(HW3) || defined(HW4) || defined(HW5) || defined(SERIAL_MONITOR))
# error !!! PLEASE CHOOSE HARDWARE VERSION !!!
#endif

//******************************************
// ENABLE MODULES
//******************************************
// remove // before #define to enable a module
#define enable_SNES
#define enable_NP
#define enable_SV
#define enable_MD
#define enable_SMS
#define enable_N64
#define enable_GBX
#define enable_NES
//#define enable_FLASH
//#define enable_FLASH16
//#define enable_PCE
#define enable_WS
//#define enable_NGP
```

- I have uploaded the sketch to the Arduino before unsoldering the fuse.

**The problem**

- When I turn it on, with either 5V or 3.3V with everything plugged in (clock generator and LCD) the Arduino just seem to reset itself in an infinite loop. I see the red "L" led flashing slowly then quickly then it repeats. There's nothing in the serial monitor at all and the serial monitor flip back between diconnected (grayed out) and connected (white) each time the arduino reset itself. I output some text at the beginning of the Setup() function but I don't even see that leading me to think that it resets before the bootloader even hands control of the code to the card reader firmware.

- Not sure it it's intended but the status led on the cart reader doesn't light up at all, I have a blue led + 220 ohm resistor.
- When I unplug EXP1 and turn on the cart reader again this time the Arduino blink the L led a couple of times and finally get to solid green. I get an error message in the serial monitor telling me it can't find the SD card.
- When I turn it on without the EXP1 connector and insert EXP1 after powering on it sometimes light up the LCD screen but there's nothing on screen.  If I turn off the power switch and turn it on again it just goes back to black as if not powered.
![20220721_013040](https://user-images.githubusercontent.com/9730015/180137166-382660e7-5e36-4aaf-b1ed-6c36e59d861f.jpg)

**Things I've tried**

- Replacing the EXP1 and EXP2 cable and vice-versa.
- Replacing the USB cable with several others.
- Tried a bunch of different usb cellphone adapter, usb wall adapter and computers.
- Resoldered the 5V wire between arduino and board a couple times. Started with Dupont but suspected a bad connection so switched to just 1 wire.
- Tried to disable some modules in the code to save some memory.
- Uploaded some blank code with everything plugged in. I get a solid green led so it seems to work fine.
- Used another LCD same brand and model without my custom 3.3V "Fix" shown above.

At this point I'm just wondering if:

1. The Arduino is faulty.
2. 5V is shaky and undervoltage is resetting the Arduino.
3. The LCD is drawing too much current forcing Arduino reset.
4. The LCD is less compatible than I thought

At this point any help would really be appreciated. I have ordered the Makerbase LCD from AliExpress but it might take a couple of weeks before it arrives.

Thank you!

**Update**

Just wired this to a regular sized Arduino Mega:

![20220721_140646](https://user-images.githubusercontent.com/9730015/180284022-35a10f3a-894a-4db3-8a75-afebc434e264.jpg)

At first I got the SD Error

<img width="659" alt="sanni_sermon" src="https://user-images.githubusercontent.com/9730015/180284177-f7135f35-1e47-4268-b024-8ecb001e738f.PNG">

Then I got it to work (almost!) by rearanging some wires. It takes maybe a good 5 seconds before power on for the LCD to turn on but it seems to consistently turn on. (Normal Behavior?)

Now I seem to have an issue with the button pressing on it's own. 95% of the time it just skips the main menu straight to the GB menu (first item). Then it stops and I can scroll and press the button normally. If I reset it goes straight to GB again.

I noticed that if I jerk the the EXP1 and or EXP2 cable a bit, it'll register a button press on it's own so I am guessing I might have a bad solder joint or bad wire connection to BTN_ENC, BTN_EN1 or BTN_EN2, I'll check that next but I am glad the screen is working even if the brand is different. 

**Update 2**

I removed most of the dupont wire keeping only the necessary connections for the LCD/SD/Buttons/LED/Clock Gen and Arduino Power I reworked some of my solder joint on the EXP1 and EXP2 connectors and corresponding Arduino Female header pins. This time it seems to be working pretty well, although navigating the menus is a bit slow (2-3 second lag when I press the button for the next menu to appear),  about 10 seconds before the LCD lights up on power on. If I move the LCD too much the button will click on its own but I believe this is due a not strong enough connection of the dupont wires and the (too long) length of the 2 ribbon cables. I am waiting for some ribbon cables and IDC connectors in the mail so I can make my own short cables. Next I'll retry with the Arduino 2560 Pro to see if it fixed my problem.

![20220721_195526](https://user-images.githubusercontent.com/9730015/180334700-8b257a4c-bed6-4e35-9689-373779bded4d.jpg)

**Update 3**

Tried again with the MEGA 2560 Pro. Same problem, I took some reading with a multimeter I am getting a steady 5V on the VDDA pins. The I/O pins output only 3V when high which is really weird... it falls in the undefined range for the atmega 2560 reset pin so that might be why it keeps resetting. The signals for LCD_EN, LCD_RST, BTN_EN1 and BTN_EN2 all give 3.0V instead of 5V so that might be why the LCD doesn't power on.  At this point I suspect the MEGA 2560 pro I got is faulty, it should generate ~5V for logic high with VCC = 5V and ~0V for logic low but that's not the case...

I also tried powering the MEGA 2560 Pro with a 300W banana clip adjustable power supply completely bypassing the USB port. I get ~3.0V for Logic high when I provide 5V Vcc and I get ~2.4V when I provide 3.3V Vcc. I tried the same experiment on an Elego (Elegoo?) Mega 2560 R3 (full size) Arduino Mega2560 Chinese clone. For the chinese clone I get 5V logic high with 5V vcc and 3.3V logic High with 3.3V Vcc. It looks like something is causing a voltage drop on the 2560 Pro I'm wondering if anybody ran into that issue before?

**Update 4**

I took a better look at the MCU chip packages and compared them with known good working mega 2560 I have. Can you guess which one is faulty?
![20220722_010753](https://user-images.githubusercontent.com/9730015/180572670-74a49277-6155-41cd-a82e-7275f0fc04b8.jpg)
![20220722_010836](https://user-images.githubusercontent.com/9730015/180572674-5b77177e-039c-4d1c-bbf2-fdca418c468f.jpg)
![20220722_010922](https://user-images.githubusercontent.com/9730015/180572675-ca5f6982-de3c-483f-b67b-30f4bade2e20.jpg)

Yep, it's the mega 2560 pro in the 3rd pic. Looks like I got a cheap bootleg low quality counterfeit chip. I have returned it for refund and I am now waiting for another (a bit more expensive but hopefully legit) mega 2560 pro. I'll post an update when I get it.

**Update 5**

I received 2 more mega 2560 pro. The atmel chips look legit this time. I tested the voltage on the reset pin and sure enough this time I get 5V for 5V and 3.3V for 3.3V. Moral of the story, don't be hasty and buy a cheap Mega 2560 pro off of Amazon. The brand was `ASHATA`, I would recommend not buying your mega 2560 pro from them. The one I got from aliexpress branded `Keyestudios` was a legit atmega chip.

Here's the final build minus the n64 port which I am still waiting for!

![20220805_114453](https://user-images.githubusercontent.com/9730015/183114773-4a8eb7a4-2828-48a8-8ac4-9f14abc82572.jpg)

**Comment by sanni** [comment]
Just a side note, you should only enable one hardware type, so either HW5 or serial monitor but not both at the same time.

````
//******************************************
// !!! CHOOSE HARDWARE VERSION !!!
//******************************************
// Remove // in front of the line with your hardware version
 #define HW5
// #define HW4
// #define HW3
// #define HW2
// #define HW1 
#define SERIAL_MONITOR
````

**Reply by VincentEmond** [reply]
Duly noted, I didn't realize those were mutually exclusive. I'll make sure to flash the new Arduino I get with only HW5 enabled. Thanks!

**Comment by sanni** [comment]
I'm happy that the Keyestudios Arduino works. Even if it's more expensive at least for now they seems to use the "old" PCB and Atmega chip that has worked for the past years.
I'm sure eventually someone will figure out what exactly got changed with the new Mega Pro PCB/Chip.

If you find the time could you make me a high resolution image of the non-working Mega Pro's PCB, both front and back?
I'm kinda curious if the PCB has changed something or if it's the chip itself.

**Reply by VincentEmond** [reply]
Unfortunately I have returned it to amazon and got refunded. I did take some pictures before but that's all I've got. 

![20220722_121609](https://user-images.githubusercontent.com/9730015/183301816-c065ff65-177d-49d2-bf93-70895c1aa316.jpg)
![20220722_121558](https://user-images.githubusercontent.com/9730015/183301817-6376761a-27c2-4a38-83ac-666848083313.jpg)
![20220722_121552](https://user-images.githubusercontent.com/9730015/183301822-1af1cd8b-56ce-43a0-88a7-e173d7cb6010.jpg)
![20220722_121534](https://user-images.githubusercontent.com/9730015/183301826-1d5adfef-4d81-427a-8452-4de96c05c2cc.jpg)
![20220722_121506](https://user-images.githubusercontent.com/9730015/183301830-2af1a88e-b7df-440a-bfce-99a425c433d1.jpg)
![20220722_121501](https://user-images.githubusercontent.com/9730015/183301834-5c97b219-a242-44c5-8542-f93ca3943b7d.jpg)

**Reply by sanni** [reply]
Those are perfect, thanks alot.

**Comment by VincentEmond** [comment]
Just got 2 more mega 2560 pro off of Amazon. A quick visual inspection of the Atmel chip suggest it is authentic. I did some testing with signal voltage and I do get a good 3.3V at 3.3V input and 5V at 5V input. This one is branded `Songhe` aka `松和` and sold by `Layger` at the top right of the board I see a `RobotDyn` inscription and it's labeled as `(EMBED)` I got 2 of them and both of them seem legit.

![20220810_131054](https://user-images.githubusercontent.com/9730015/183976669-3cbf207f-4e65-4780-a579-6249cdcf8a2f.jpg)
![20220810_131035](https://user-images.githubusercontent.com/9730015/183976673-ec15b4e4-941c-4380-8416-580f4ffd99f2.jpg)
![20220810_131007](https://user-images.githubusercontent.com/9730015/183976677-d192f543-51c9-4f0c-9c06-53fcf80b3d42.jpg)
![20220810_130923](https://user-images.githubusercontent.com/9730015/183976678-bd579ce8-25c2-46d0-acf2-45d6939c81cd.jpg)
